lunes_cms/cms/migrations/0007_document_created_by_link.py

```diff
@@ -0,0 +1,74 @@
+# Generated by Django 3.2.13 on 2022-07-05 11:32
+
+from django.db import migrations, models
+import django.db.models.deletion
+
+
+# pylint: disable=unused-argument
+def link_created_by(apps, schema_editor):
+    """
+    Convert the ``created_by`` char field to a foreign key
+
+    :param apps: The configuration of installed applications
+    :type apps: ~django.apps.registry.Apps
+
+    :param schema_editor: The database abstraction layer that creates actual SQL code
+    :type schema_editor: ~django.db.backends.base.schema.BaseDatabaseSchemaEditor
+    """
+    Document = apps.get_model("cms", "Document")
+    Group = apps.get_model("auth", "Group")
+    for document in Document.objects.filter(created_by__isnull=False):
+        group, created = Group.objects.get_or_create(name=document.created_by)
+        document.created_by_link = group
+        document.save()
+
+
+# pylint: disable=unused-argument
+def reverse_created_by(apps, schema_editor):
+    """
+    Convert the ``created_by`` field from foreign key to a char field
+
+    :param apps: The configuration of installed applications
+    :type apps: ~django.apps.registry.Apps
+
+    :param schema_editor: The database abstraction layer that creates actual SQL code
+    :type schema_editor: ~django.db.backends.base.schema.BaseDatabaseSchemaEditor
+    """
+    Document = apps.get_model("cms", "Document")
+    Group = apps.get_model("auth", "Group")
+    for document in Document.objects.filter(created_by__isnull=False):
```

Comment:
```suggestion
    for document in Document.objects.filter(created_by_link__isnull=False):
```
I just tried to reverse the migration and noticed that all individual documents get lost - there was a small bug in my suggestion.
In the reverse function, we have to assume that the migration steps are executed in the reverse order, so at the time the function runs, the database field which contains the `created_by` information is still called `created_by_link`, so the filter also has to check this field.